### PR TITLE
Added validations to make sure user can only select version from valid list according to profile selected

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Make sure the user can only select the valid version for the selected profile.

* Added js file to dataset admin so version dropdown changes when user change profile
* Added new url to give versions for a specific profile
* Added new view and serializer for the version to support js file 
* Updated code in dataset admin form to:
    * Update version queryset when a profile is changed or according to currently selected profile
    * Added validation so user can not select a version from any other profile's hierarchy

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-269

## How to test it locally
**Initial form :**
Initial form is when you try to add a new object from the admin panel
Things to check:
* If there is only one profile:
     * profile should be selected by default
     * versions should be filtered according to the selected profile
     * if there is only one version select that version automatically
* If there are multiple profiles:
    * Profile is not selected but a user can only select a profile that exists in user groups
    * version field is empty on initial load
    * version field should be changed/populated if profile is changed
* When trying to save:
   * profile is required
   * version is required
   * If an error is thrown of version is required, version field data should be filtered according to selected profile
   * User not able to select any other version
   
**Change Form**

* Verison field is filtered according to the selected profile
* If profile field is changed version field should autoload
* if the user tries to save a profile with invalid version validations are thrown

Dataset admin: https://staging.wazimap-ng.openup.org.za/admin/datasets/dataset/

## Changelog
* Updated dataset admin form
* Updated geography hierarchy to add a new property get_version_names
* Added new view, url & serializer for versions according to profile
* Added js code to change versions when profile is changed in admin panel
* Added tests

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [x]  black was run locally (as part of the pre-commit hook)

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

**Review by michaelglenister — APPROVED** [review]
Some small spelling fixes otherwise it looks good, thanks

File: wazimap_ng/datasets/models/geography.py [hunk 1]
```diff
@@ -106,6 +106,14 @@ class GeographyHierarchy(BaseModel, SimpleHistory):
     def default_version(self):
         return self.configuration.get("default_version")
 
+    @property
+    def get_version_names(self):
+        versions = self.configuration.get("versions", [])
```
[the remark this answers is not in the document]
Comment (goyal1092) [reply wazimap_ng/datasets/models/geography.py:111]: @michaelglenister It's a JSON field so while setting up a profile there can be an instance when we do have that set yet.
